src/GeoVector.h

```diff
@@ -39,97 +39,101 @@ class GeoVector {
   inline GeoVector<T> operator/(const T other) const;
 
   void rotateToAngleAroundPoint(const T angleRotationValue, const GeoVector<T>& point);
+
 private:
-  T _x;
-  T _y;
+  GeoVector(glm::vec<2, T> value);
+
+  inline glm::vec<2, T> getValue() const;
+
+  inline void setValue(glm::vec<2, T> value);
+
+  glm::vec<2, T> _value;
 };
 
 template<typename T>
-GeoVector<T>::GeoVector(T x, T y) {
-  _x = x;
-  _y = y;
+GeoVector<T>::GeoVector(T x, T y) : _value(glm::vec2(x, y)) {
+}
+
+template<typename T>
+GeoVector<T>::GeoVector(glm::vec<2, T> value) : _value(value) {
 }
 
 template<typename T>
 T GeoVector<T>::getX() const {
-  return _x;
+  return getValue().x;
 }
 
 template<typename T>
 void GeoVector<T>::setX(T value) {
-  GeoVector::_x = value;
+  _value.x = value;
```

Comment:
This, and the `setY` method are the only places still accessing `_value` directly.

I don't think its desirable to construct a new vector, setting both x and y in this case; since the code would need to look like (or similar to):
```cpp
auto vec = getValue();
vec.x = value;
setValue(vec);
```

Reply:
Maybe if it returned a ref instead?

Reply:
Yep im happy with that.